### PR TITLE
feat(execute.go): add interface for a mixin to suppress output

### PR DESCRIPTION
**Pull request by vdice**
# What does this change
* Adds interface to allow mixins to suppress step/command output
* Adds implementation for `exec` mixin

# What issue does it fix
Contributes towards https://github.com/deislabs/porter/issues/940

# Notes for the reviewer

* Do we also want to add support to the `kubernetes` mixin in this repo?

* Docs: Where should mention of this step configuration be added?  Or file a follow-up ticket once most mixins have implemented support?

* Can refactor the added integration test to align with updates in https://github.com/deislabs/porter/pull/953 (moving to using script(s) for `exec` steps, etc.)

# Checklist
- [x] Unit Tests
- [ ] Documentation
  - [ ] Documentation Not Impacted

### CODE REVIEW

File: pkg/exec/builder/execute.go [hunk 1]
```diff
@@ -83,15 +87,27 @@ func ExecuteStep(cxt *context.Context, step ExecutableStep) (string, error) {
...
+	if suppressable, ok := step.(SuppressesOutput); ok {
+		if suppressable.SuppressesOutput() {
+			out = nil
+			if cxt.Debug {
+				fmt.Fprintf(cxt.Err, "DEBUG: output suppressed for command %s\n", prettyCmd)
+			}
+		}
+	}
+	cmd.Stdout = out
+	cmd.Stderr = cxt.Err
```
Comment (carolynvs) [inline pkg/exec/builder/execute.go]: With suppress output we should use ioutil.Discard for both stdout and stderr.

File: pkg/exec/builder/execute.go [hunk 2]
```diff
@@ -83,15 +87,27 @@ func ExecuteStep(cxt *context.Context, step ExecutableStep) (string, error) {
...
 		fmt.Fprintln(cxt.Out, prettyCmd)
 	}
 
+	// Setup output stream for command
+	// If Step suppresses output, update stream accordingly
+	output := &bytes.Buffer{}
+	out := io.MultiWriter(cxt.Out, output)
+	if suppressable, ok := step.(SuppressesOutput); ok {
+		if suppressable.SuppressesOutput() {
+			out = nil
```
[the remark this answers is not in the document]
Comment (vdice) [reply pkg/exec/builder/execute.go]: Ah yes, thank you.  Updates incoming...

File: pkg/exec/builder/execute.go [hunk 3]
```diff
@@ -83,15 +87,27 @@ func ExecuteStep(cxt *context.Context, step ExecutableStep) (string, error) {
 	args = splitCommand(args)
 
 	cmd := cxt.NewCommand(step.GetCommand(), args...)
-	output := &bytes.Buffer{}
-	cmd.Stdout = io.MultiWriter(cxt.Out, output)
-	cmd.Stderr = cxt.Err
 
 	prettyCmd := fmt.Sprintf("%s%s", cmd.Dir, strings.Join(cmd.Args, " "))
```
[the remark this answers is not in the document]
Comment (vdice) [reply pkg/exec/builder/execute.go]: I moved this to the appropriate conditional (if suppressOutput is false)... WDYT?

File: pkg/exec/builder/execute.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (vdice) [reply pkg/exec/builder/execute.go]: Updated.

**Pull request merged**